app/models/manageiq/providers/vmware/infra_manager/storage.rb

```diff
@@ -1,3 +1,12 @@
 class ManageIQ::Providers::Vmware::InfraManager::Storage < ManageIQ::Providers::InfraManager::Storage
   include ManageIQ::Providers::Vmware::InfraManager::EmsRefObjMixin
+
+  supports :smartstate_analysis do
+    if nil?
+      unsupported_reason_add(:smartstate_analysis, "Storage not found")
+    end
+    unless storage_type_supported_for_ssa?
+      unsupported_reason_add(:smartstate_analysis, "Smartstate Analysis unsupported for storage type %{store_type}" % {:store_type => store_type})
+    end
+  end
```

Comment:
I think this needs to live in `app/models/manageiq/providers/vmware/infra_manager/vm_or_template_shared/scanning.rb` (where `storage.nil?` would make sense, just `nil?` here doesn't do anything since the current object can't be nil here)